### PR TITLE
chore(docs): update nextra

### PR DESCRIPTION
**Pull request by tknickman**
### Description

Needed to fix the missing copy code button on the docs, ended up updating Nextra and Next while I was in there. 

### Testing Instructions

Checkout the preview!

### CODE REVIEW

**Review by arlyon — APPROVED** [review]

File: .npmrc [hunk 1]
```diff
@@ -1,2 +1,2 @@
 auto-install-peers = true
-hoist = false
+hoist-pattern[]=vfile-message
```
Comment (tknickman) [inline .npmrc:2]: `rehype-katex` was added to `nextra` in 2.0.2 to support LaTeX which brings in `hast-util-from-html` which has a dependency issue. 

I sent them a PR here: https://github.com/syntax-tree/hast-util-from-html/pull/4, but need to patch it on our end for now.

File: docs/next.config.js [hunk 2]
```diff
@@ -1,9 +1,8 @@
 const { withSentryConfig } = require("@sentry/nextjs");
 const withNextra = require("nextra")({
   theme: "nextra-theme-docs",
-  themeConfig: "./theme.config.js",
-  unstable_flexsearch: true,
-  unstable_staticImage: true,
+  themeConfig: "./theme.config.tsx",
+  defaultShowCopyCode: true,
```
Comment (tknickman) [inline docs/next.config.js:5]: I don't know why this isn't set by default, have to check on that one (cc @shuding)

File: docs/pages/blog/_meta.json [hunk 3]
```diff
@@ -1,6 +1,5 @@
 {
   "*": {
-    "//": "Hide all the links from the sidebar or navbar, and disable ToC and Sidebar for these pages.",
```
Comment (tknickman) [inline docs/pages/blog/_meta.json:3]: New nextra isn't a fan of this key in meta (jsonc support wen)

File: docs/theme.config.tsx [hunk 4]
```diff
@@ -9,12 +9,9 @@ import { Discord, Github } from "./components/Social";
 
 const SITE_ROOT = "https://turbo.build";
 
-/**
- * @type {import('nextra-theme-docs').DocsThemeConfig}
- */
-const theme = {
+const config: DocsThemeConfig = {
```
Comment (tknickman) [inline docs/theme.config.tsx:12]: Thank god we have real types here now!

File: docs/next.config.js (hunk 2 above)
Comment (shuding) [reply docs/next.config.js:5]: Will make that the default!

**Pull request merged**